src/Nancy/ErrorHandling/DefaultStatusCodeHandler.cs

```diff
@@ -62,47 +66,60 @@ public void Handle(HttpStatusCode statusCode, NancyContext context)
                 return;
             }
 
-            string errorPage;
-
-            if (!this.errorPages.TryGetValue(statusCode, out errorPage))
+            if (!this.errorMessages.ContainsKey(statusCode) || !this.errorPages.ContainsKey(statusCode))

```

Comment:
You're right, this protects against a status that doesn't have a message and an error page. I was referring to calling code that doesn't check `HandlesStatusCode` before calling `Handle`. Then someone could inadvertently pass in a non-supported status.